pkg/catalogv2/system/system.go

```diff
@@ -427,9 +432,10 @@ func (m *Manager) isInstalled(namespace, name, version, minVersion string, desir
 			}
 		}
 
-		if (desired.LessThan(ver) || desired.Equal(ver)) && bytes.Equal(patchedJSON, actualValueJSON) {
-			return true, "", nil, nil
+		if !desired.Equal(ver) {
```

Comment:
1. I think the old check should be used if minVersion is specified because, without it we are forcing other charts to always be on the latest release.
2. In this new check we need to also validate that we are installed with the expected values.

Reply:
1. I don't believe so, since `isInstalled` receives the minVersion parameter. It's all about the value that the caller would pass in. In our case, `install` determines what `minVersion` means.

Reply:
I'm also a bit confused on the behavior here. In the following scenario:
- Min version is 2.0.1
- Installed is 2.0.2
- Exact version is "" (empty)

I would expect isInstalled to return `true`. However, this code seems to indicate that this would now return false, which may cause charts which use the min version to be re-installed. Would you mind clarifying this a bit more?

Reply:
It would be helpful to add a unit test just for the `isInstalled` function. That way, we can make sure we cover all of the edge-cases, but don't have to refactor the entire file. If we need to, we can even break up the function and just test the logic we are changing.